bigquery/src/main/java/org/apache/iceberg/gcp/bigquery/BigQueryMetastoreCatalog.java

```diff
@@ -107,6 +107,28 @@ public void initialize(String name, Map<String, String> properties) {
     initialize(name, properties, projectId, projectLocation, client);
   }
 
+  @VisibleForTesting
+  BigQueryClientFactory createClientFactory(Map<String, String> properties) {
+    String factoryClassName = properties.get(CLIENT_FACTORY);
+    BigQueryClientFactory factory;
+
+    if (factoryClassName != null) {
+      try {
+        LOG.info("Loading custom BigQuery client factory: {}", factoryClassName);
+        Class<?> factoryClass = Class.forName(factoryClassName);
```

Comment:
I don't feel like this is the right way trigger the different behavior between default/impersonation.  Reflection based injection is generally reserved for where it makes sense to extend for custom implementations, but this is largely about how to configure the client options.